dev/RadioButtons/RadioButtons.cpp

```diff
@@ -7,136 +7,634 @@
 #include "Vector.h"
 #include "RuntimeProfiler.h"
 #include "ResourceAccessor.h"
+#include "RadioButtonsTestHooks.h"
 
 RadioButtons::RadioButtons()
 {
     __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadioButtons);
 
-    auto items = winrt::make<Vector<winrt::IInspectable>>();
+    m_selectionModel = winrt::SelectionModel{};
+    m_selectionModel.SingleSelect(true);
+    m_selectionModel.Source(GetItemsSource());
+    m_selectionChangedRevoker = m_selectionModel.SelectionChanged(winrt::auto_revoke, { this, &RadioButtons::OnSelectionChanged });
+
+    auto const items = winrt::make<Vector<winrt::IInspectable>>();
     SetValue(s_ItemsProperty, items);
 
     SetDefaultStyleKey(this);
+
+    // Override normal up/down/left/right behavior -- down should always go to the next item and up to the previous.
+    // left and right should be spacial but contained to the RadioButtons control. We have to attach to PreviewKeyDown
+    // because RadioButton has a key down handler for up and down that gets called before we can intercept.
+    PreviewKeyDown({ this, &RadioButtons::OnChildKeyDown });
+    GettingFocus({ this, &RadioButtons::OnGettingFocus });
+    GotFocus({ this, &RadioButtons::OnChildGotFocus });
+    KeyDown({ this, &RadioButtons::KeyDownHandler });
+    KeyUp({ this, &RadioButtons::KeyUpHandler });
+
+    s_childHandlersProperty =
```

Comment:
Please add a note here. It might not be obvious why this attached property is needed.